src/main/java/org/benetech/servicenet/service/impl/DataStatusServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package org.benetech.servicenet.service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.benetech.servicenet.domain.Organization;
+import org.benetech.servicenet.domain.SystemAccount;
+import org.benetech.servicenet.repository.OrganizationRepository;
+import org.benetech.servicenet.repository.SystemAccountRepository;
+import org.benetech.servicenet.service.DataStatusService;
+import org.benetech.servicenet.service.dto.DataStatusDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service Implementation for managing DataStatus.
+ */
+@Service
+public class DataStatusServiceImpl implements DataStatusService {
+
+    @Autowired
+    private SystemAccountRepository systemAccountRepository;
+
+    @Autowired
+    private OrganizationRepository organizationRepository;
+
+    @Override
+    public Page<DataStatusDto> getDataStatuses(Pageable pageable) {
+        List<DataStatusDto> results = new ArrayList<>();
+        Page<SystemAccount> systemAccounts = systemAccountRepository.findAll(pageable);
+        for (SystemAccount systemAccount : systemAccounts) {
+            organizationRepository.findFirstByAccountIdOrderByUpdatedAtDesc(systemAccount.getId())
```

Comment:
Shouldn't we actually look at data import reports? I believe that organization's updated at can be modified not only by the imports